worker/transfer.go

```diff
@@ -52,64 +52,59 @@ func parallelUploadSlab(ctx context.Context, shards [][]byte, hosts []sectorStor
 		root types.Hash256
 		err  error
 	}
-	reqChan := make(chan req, len(shards))
-	defer close(reqChan)
-	respChan := make(chan resp, len(shards))
-	worker := func() {
-		for r := range reqChan {
-			doneChan := make(chan struct{})
-
-			// Trace the upload.
-			ctx, span := tracing.Tracer.Start(ctx, "upload-request")
-			span.SetAttributes(attribute.Stringer("host", r.host.PublicKey()))
-			span.SetAttributes(attribute.Stringer("contract", r.host.Contract()))
-
-			go func(req req) {
-				defer close(doneChan)
-
-				lockID, err := locker.AcquireContract(ctx, req.host.Contract(), contractLockingUploadPriority, 30*time.Second)
-				if err != nil {
-					respChan <- resp{req, types.Hash256{}, err}
-					span.SetStatus(codes.Error, "acquiring the contract failed")
-					span.RecordError(err)
-					return
-				}
-				defer locker.ReleaseContract(ctx, req.host.Contract(), lockID)
+	respChan := make(chan resp, 2*len(hosts)) // every host can send up to 2 responses
```

Comment:
Avoiding a weird edge case where we are done uploading but due to slow hosts we have sent so many requests to workers that they can no longer push them in the response channel without a goroutine emptying it.

Reply:
That's a bug I introduced with adding timeouts, thanks.